ccvm_simulators/ccvmplotlib/problem_metadata/boxqp_metadata.py

```diff
@@ -80,16 +80,17 @@ def ingest_metadata(self, metadata_filepath: str) -> None:
         self.__batch_size = self.__df["batch_size"][0]
         self.__problem_size_list = sorted(self.__df["problem_size"].unique().tolist())
 
-    def generate_TTS_plot_data(
+    def generate_plot_data(
         self,
-        machine_time_func: callable,
+        metric_func: callable,
     ) -> pd.DataFrame:
         """Calculate the time to solution vs problem size for a particular gap and
         quantile.
 
         Args:
-            machine_time_func (callable): A callback function that calculates the
-            machine time, which is used to compute the TTS.
+            metric_func (callable): A callback function that calculates a metrics.
+            It can be either machine time or energy max calculating function, which
+            is used to compute the TTS and ETS, respectively.
```

Comment:
I'm a little concerned at calling it this because I think there's more that goes into the metrics calculation than this function, this function just calculates a component of it. I don't have a good alternative right now, let me think on it.

For this reason, I think this description might be more accurate:

```suggestion
            metric_func (callable): A callback function that is used when calculating the metrics either to determine the `machine_time` or the `energy_max`, which are used when computing the TTS or ETS, respectively.
```

Reply:
I'm okay with calling it `metric_func` with this definition that you've given, @KatieO4t1qbit . I know the word "metric" may not be the best choice but I can't think of anything else right now.

Reply:
Ya, I still have no better ideas either. `metric_func` it is!